File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/LocaleManager.java
package com.esafirm.imagepicker.helper;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Build;

import java.util.Locale;

public class LocaleManager {

    private static String language;

    public static void setLanguange(String newLanguage) {
        language = newLanguage;
    }

    private static String getLanguage() {
        return language != null && !language.isEmpty()
                ? language
                : Locale.getDefault().getLanguage();
    }

    public static Context updateResources(Context context) {
        Locale locale = new Locale(getLanguage());
        Locale.setDefault(locale);

        Resources res = context.getResources();
        Configuration config = new Configuration(res.getConfiguration());
        if (Build.VERSION.SDK_INT >= 17) {
            config.setLocale(locale);
            context = context.createConfigurationContext(config);
        } else {
            config.locale = locale;
            res.updateConfiguration(config, res.getDisplayMetrics());
        }
        return context;
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/LocaleManager.kt
package com.esafirm.imagepicker.helper

import android.content.Context
import android.content.res.Configuration
import android.os.Build
import java.util.Locale

object LocaleManager {

    var language: String = Locale.getDefault().language

    fun updateResources(context: Context): Context {
        var resultContext = context
        val locale = normalizeLocale(Locale(language))

        Locale.setDefault(locale)

        val res = resultContext.resources
        val config = Configuration(res.configuration)

        config.setLocale(locale)
        resultContext = resultContext.createConfigurationContext(config)

        return resultContext
    }

    private fun normalizeLocale(localeLanguage: Locale): Locale {
        val ZH = "zh"
        val TW = "TW"
        val CN = "CN"
        val locale: Locale
        val newLocaleLanguage = localeLanguage.toString()
        return when {
            newLocaleLanguage.length == 5 -> {
                locale = Locale(
                    newLocaleLanguage.substring(0, 2),
                    newLocaleLanguage.substring(3, 5).toUpperCase(Locale.getDefault())
                )
                locale
            }

            newLocaleLanguage == ZH -> {
                locale = if (Locale.getDefault().country == TW) {
                    Locale(ZH, TW)
                } else {
                    Locale(ZH, CN)
                }
                locale
            }

            else -> {
                localeLanguage
            }
        }
    }
}
